app/models/tracepoint.rb

```diff
@@ -39,4 +39,10 @@ def to_xml_node(print_timestamp = false)
     el1 << (XML::Node.new("time") << timestamp.xmlschema) if print_timestamp
     el1
   end
+
+  # Return points of trackable traces in original order
+  scope :trackable_ordered, -> { joins(:trace).where(:gpx_files => { :visibility => %w[trackable identifiable] }).order("gpx_id DESC, trackid ASC, timestamp ASC") }
+
+  # Hide the order of points of non-trackable traces for privacy
+  scope :non_trackable_unordered, -> { joins(:trace).where(:gpx_files => { :visibility => %w[public private] }).order("gps_points.latitude", "gps_points.longitude") }
```

Comment:
Can you move these to the top please, for consistency with other models - normally we put scopes immediately after the associations at the top of the model.

As to names I agree with @gravitystorm that these names may be confusing but I'm not sure the ones I suggested are perfect either so I'm not really sure what's best. The goal of my names was just to separate point which could be ordered (as `ordered`) from those which can't (as `unordered`) but I realise that may not be very clear from the names.

Reply:
Ah, I see I missed the model format earlier. Yes, I moved the new scopes to the right location in the model now.

I'm not sure what to suggest for names.  As a newcomer it makes sense to me that identifiable and trackable traces both have the property trackable?==true and are both included in a trackable scope.  `unordered` is trickier---perhaps the ones that have trackable?==false could be called `anonymized` or `non_trackable_anonymized` or just `non_trackable`?  But as I study it more, "public" traces still puzzle me---I don't see the value of anonymizing the points on the map while the trace remains publicly available for download.  I'm guessing this has already been discussed elsewhere.